### DISCUSSION TITLE
[PHOEBE 2.4] Scaling Synthetic Flux Values With Real Flux

### DISCUSSION THREAD

**Discussion by rfreire-400** [phoebe-project/phoebe2#732] [Questions]
Hello,

I'm trying to plot the synthetic flux values with the real ones but it's way off. Does anyone know how to fix this?

![Question for Gith](https://github.com/phoebe-project/phoebe2/assets/113457579/4e353898-7017-4056-8ae2-51b3337c7a44)

Synthetic flux in blue, real flux in black

**Comment by aprsa** [comment]
It looks like the means of your synthetic fluxes and your observed fluxes match up pretty well; the problem is that your model is way off. As you change other parameters to more suitable values, automatic scaling will still keep the means aligned.

**Reply by rfreire-400** [reply]
Is there any way to manually change the scalling? I'm studying a system with close to no available data on and I have added everything I can

**Reply by aprsa** [reply]
You can, see [this tutorial](http://phoebe-project.org/docs/latest/tutorials/pblum) for details; however, I do not think that is not your problem. I believe that your equivalent radii, inclination and effective temperatures are way off.

**Reply by rfreire-400** [reply]
All right, thanks! I'll try to find more info about it and play around with it a bit more.

In the mean time, I was having some different issues with the code.
So whenever I want to run `b.set_value_all('atm', 'blackbody')`, it gives me an error saying that the parameter does not exist ("ValueError: no parameters found"). Then, when I try to run `b.run_compute()`, it spits out an error saying how the parameter set for atm is invalid. And after I rerun `b.set_value_all('atm', 'blackbody')` again, it finally fixes itself and runs fine.

Is there any way to fix this? It seems that the atm parameter is only created when I try to run_compute()

**Reply by aprsa** [reply]
Which version of phoebe are you using? I just tried this and it works:
```
import phoebe
b = phoebe.default_binary()
b.set_value_all('atm', 'blackbody')
```
Does this not work for you?

**Reply by rfreire-400** [reply]
Never mind, sorry. I think I fixed it. Basically, here's what happened

I'm using Phoebe 2.4.10. Here's the relevant part of the code:
```
b['period@binary'] = period  
b['per0@binary'] = 160.5
b['sma@binary'] = sma 

b.add_dataset('lc', fluxes=flux, times=time, sigmas=flux_err, dataset='lc01', overwrite=True)

b['teff@primary@component'] = 15668
b['requiv@primary@component'] = 6

b['gravb_bol@primary'] = 1.0
b['irrad_frac_refl_bol@primary'] = 1.0

b.set_value_all('ld_mode', 'manual')
b.set_value_all('ld_mode_bol', 'manual')
b.set_value_all('atm', 'blackbody')
b.set_value('pblum_mode', 'dataset-scaled')
```

This would give me: "ValueError: no parameters found".
'b.run_compute()' would not work after running this. It was only after failing to run_compute() that I could do 'b.set_value_all('atm', 'blackbody')' and then everything else would work as intended.

Essentially, I had to run: chunk of code -> 'b.run_compute()' -> chunk of code -> 'b.run_compute()' and only after doing that everything else would work. Removing 'b.set_value_all('atm', 'blackbody')' would also give an error when running b.run_compute().

To fix it, I added a mass=6.2 for primary and then run_compute() would run without issues. It still gives an error for 'b.set_value_all('atm', 'blackbody')' but it's able to run the code without it so I just ended up removing it.

Sorry for the wall of text

**Reply by aprsa** [reply]
I still don't see anything obvious in the script that would trip `set_value_all()`; can you please send a *minimal* working example that demonstrates the problem?

**Reply by rfreire-400** [reply]
<img width="797" alt="Screenshot 2023-05-10 at 9 15 54 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/788b92d9-3795-44fc-b680-1f0a5b66b673">
<img width="797" alt="Screenshot 2023-05-10 at 9 16 07 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/a0951719-51b5-4a44-b8b4-f0fbd66965e1">

run_compute() works fine after running this, I just get this annoying error message

**Reply by kecnry** [reply]
`phoebe.Bundle()` does not come with compute-options (which is where the atm parameter lives).  If you also add a `b.add_compute('phoebe')` when building your own binary, or use `b = phoebe.default_binary()` instead of building your own, then it should be available immediately.

**Reply by rfreire-400** [reply]
I understand. Thanks a lot!
